contrib/example-backend/controllers/clusterbinding/clusterbinding_controller.go

```diff
@@ -19,300 +19,237 @@ package clusterbinding
-			c.enqueueServiceExport(logger, newObj)
-		},
-		DeleteFunc: func(obj any) {
-			c.enqueueServiceExport(logger, obj)
-		},
-	}); err != nil {
-		return nil, err
-	}
-
-	return c, nil
+	return r, nil
 }
 
-type Resource = committer.Resource[*kubebindv1alpha2.ClusterBindingSpec, *kubebindv1alpha2.ClusterBindingStatus]
-type CommitFunc = func(context.Context, *Resource, *Resource) error
-
-// Controller reconciles ClusterBinding conditions.
-type Controller struct {
-	queue workqueue.TypedRateLimitingInterface[string]
-
-	clusterBindingLister  bindlisters.ClusterBindingLister
-	clusterBindingIndexer cache.Indexer
-
-	serviceExportLister  bindlisters.APIServiceExportLister
-	serviceExportIndexer cache.Indexer
-
-	clusterRoleLister  rbaclisters.ClusterRoleLister
-	clusterRoleIndexer cache.Indexer
-
-	clusterRoleBindingLister  rbaclisters.ClusterRoleBindingLister
-	clusterRoleBindingIndexer cache.Indexer
-
-	namespaceLister  corelisters.NamespaceLister
-	namespaceIndexer cache.Indexer
-
-	reconciler
-
-	commit CommitFunc
-}
-
-func (c *Controller) enqueueClusterBinding(logger klog.Logger, obj any) {
-	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
-	if err != nil {
-		runtime.HandleError(err)
-		return
-	}
-
-	logger.V(2).Info("queueing ClusterBinding", "key", key)
-	c.queue.Add(key)
-}
-
-func (c *Controller) enqueueServiceExport(logger klog.Logger, obj any) {
-	seKey, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
-	if err != nil {
-		runtime.HandleError(err)
-		return
-	}
-	ns, _, err := cache.SplitMetaNamespaceKey(seKey)
-	if err != nil {
-		runtime.HandleError(err)
-		return
-	}
-
-	key := ns + "/cluster"
-	logger.V(2).Info("queueing ClusterBinding", "key", key, "reason", "APIServiceExport", "ServiceExportKey", seKey)
-	c.queue.Add(key)
-}
-
-// Start starts the controller, which stops when ctx.Done() is closed.
-func (c *Controller) Start(ctx context.Context, numThreads int) {
-	defer runtime.HandleCrash()
-	defer c.queue.ShutDown()
-
-	logger := klog.FromContext(ctx).WithValues("controller", controllerName)
-
-	logger.Info("Starting controller")
-	defer logger.Info("Shutting down controller")
-
-	for i := 0; i < numThreads; i++ {
-		go wait.UntilWithContext(ctx, c.startWorker, time.Second)
-	}
-
-	<-ctx.Done()
-}
-
-func (c *Controller) startWorker(ctx context.Context) {
-	defer runtime.HandleCrash()
-
-	for c.processNextWorkItem(ctx) {
+// mapServiceExportToClusterBinding maps APIServiceExport events to ClusterBinding reconcile requests.
+func mapServiceExportToClusterBinding(ctx context.Context, obj client.Object) []reconcile.Request {
+	// Extract namespace from the APIServiceExport
+	serviceExport := obj.(*kubebindv1alpha2.APIServiceExport)
+
+	// The original logic created a ClusterBinding key as "namespace/cluster"
+	return []reconcile.Request{
+		{
+			NamespacedName: types.NamespacedName{
+				Namespace: serviceExport.Namespace,
+				Name:      "cluster", // This matches the original logic: ns + "/cluster"
```

Comment:
This comment might be less useful after this is merged, maybe it should just be a review comment. Same for other "original logic" comments.